Library/Homebrew/extend/os/linux/extend/ENV/super.rb

```diff
@@ -46,15 +46,30 @@ def homebrew_extra_paths
 
   def homebrew_extra_isystem_paths
     paths = []
-    # Add paths for GCC headers when building against glibc@2.13 because we have to use -nostdinc.
-    if deps.any? { |d| d.name == "glibc@2.13" }
+    # Add paths for GCC headers when building against glibc@2.13 or glibc@2.17
+    # because we have to use -nostdinc.
+    versioned_glibcs = ["glibc@2.13", "glibc@2.17"]
+    if deps.any? { |d| versioned_glibcs.include?(d.name) }
       gcc_include_dir = Utils.safe_popen_read(cc, "--print-file-name=include").chomp
       gcc_include_fixed_dir = Utils.safe_popen_read(cc, "--print-file-name=include-fixed").chomp
       paths << gcc_include_dir << gcc_include_fixed_dir
     end
     paths
   end
 
+  sig { returns(String) }
+  def determine_optflags
+    if effective_arch == :armv8 &&
+       compiler.match?(GNU_GCC_REGEXP) &&
+       DevelopmentTools.gcc_version(compiler.to_s) >= "9.3.1"
+      # Out-of-line atomics are not supported out-of-the-box on all systems.
+      # https://learn.arm.com/learning-paths/servers-and-cloud-computing/lse/intro/
+      "#{generic_determine_optflags} -mno-outline-atomics"
```

Comment:
Given we already use `-march=armv8-a`, is this for scenarios with a shared libgcc is used?
Is this also relevant for anything outside of Portable Ruby?

Reply:
This is not related to shared libgcc AFAIK; it's relevant for building all bottles with GCC >= 9.3.1 (where out-of-line atomics are enabled by default). I recall not being able to run binaries from a bottle built on standard CI environment (GCC 11) out-of-the-box on Ubuntu 18.04 (requires an extra package `libc6-lse` to be installed).

Reply:
That's interesting given https://community.arm.com/arm-community-blogs/b/tools-software-ides-blog/posts/making-the-most-of-the-arm-architecture-in-gcc-10 implies it can determine it at runtime using a libgcc call of `__aarch64_cas4_relax`.

Is it elsewhere on 18.04?

Reply:
Mmm, will need to check this. Away from keyboard at the moment but will do tonight.

Reply:
Unless we need this to build Portable Ruby on the GitHub runner: I'd prefer if we didn't have this. Feels like something that should be formula specific.

Reply:
Welp, it's been a while since I worked on this (https://github.com/ZhongRuoyu/homebrew-portable-ruby-aarch64-linux/commit/c024aefbb35378035d4ea7d21b40e0b8eb065f1f) and now I couldn't really reproduce what I encountered (and those `__aarch64_*` calls are indeed present). But since everything seemed to work, I guess we can leave this out for now.